### ISSUE TITLE
Interface to easily programmatically score many models locally on public benchmarks

### ISSUE THREAD

**Issue by YudiXie** [brain-score/vision#611]
After the update, it seems that brain-score deprecated the interface to easily programmatically score many models locally on public benchmarks. This is essentially the function achieved by this [tutorial](https://github.com/brain-score/candidate_models/blob/master/examples/score-model.ipynb) but now it isdeprecated.

As a user of brain score, I think this use case is still very desirable because it is often the case that users want to do some quick prototyping of many different models (could be the same model architecture but trained with different hyperparameters, etc.) on a small number of public benchmarks before submitting to the brain-score full benchmark. When doing quick prototyping, users may want to keep the code for model definition (defining pytorch models, loading weights, etc.) in their own repository instead of registering the model in the brainscore_vision/models (writing all the init.py files, etc.) each time they want to score a new model with perhaps just a small variation, because this could be tedious and time-consuming. It would be great if brain score could still support this [use case](https://github.com/brain-score/candidate_models/blob/master/examples/score-model.ipynb) or perhaps use some script to automate model registration process (writing all the init.py files in brainscore_vision/models, etc.) while allowing users to keep the model definition in their own repo.

I figured out a little hack to make it work though, but this may not be ideal since it bypasses many new features such as dependency check and score saving. (https://github.com/brain-score/vision/issues/533#issuecomment-1937820878)

**Comment by mschrimpf** [comment]
I'm guessing the ideal would be either
```
from brainscore_vision import score, model_registry

model_registry['mymodel'] = ...

myscore = score(model_identifier='mymodel', benchmark_identifier=...)
```
or 
```
from brainscore_vision import score

myscore = score(model_identifier='mymodel', model=mymodel, benchmark_identifier=...)
```
Is this right?

@kvfairchild from how [`locate_plugin`](https://github.com/brain-score/core/blob/0888d8a3bad8cae561957cc95b5fe692b477b696/brainscore_core/plugin_management/import_plugin.py#L34) works, option 1 seems quite difficult to implement, is that right?

**Comment by YudiXie** [comment]
Yes, either of them would be great, and what I have in mind is that this should work in user's own repository after installing brainscore.

**Comment by kvfairchild** [comment]
> @kvfairchild from how [locate_plugin](https://github.com/brain-score/core/blob/0888d8a3bad8cae561957cc95b5fe692b477b696/brainscore_core/plugin_management/import_plugin.py#L34) works, option 1 seems quite difficult to implement, is that right?

It could be done but I think option 2 is cleaner and makes more sense for the use case. A lot of the functionality in `ImportPlugin` and `CondaScore` is to support the leaderboard scoring and not really necessary for this type of local development, so the simplest solution would be to bypass those altogether if a model is passed directly to score.

**Comment by mschrimpf** [comment]
could this then not just be as simple as
```
def local_score(model: BrainModel, benchmark_identifier: str) -> Score:
    benchmark = load_benchmark(benchmark_identifier)
    score = benchmark(model)
    return score
```
?

**Comment by mschrimpf** [comment]
Is the pain point maybe in wrapping the model? We could add an example for doing this, e.g.
```
from brainscore_vision.model_helpers.activations.pytorch import PytorchWrapper
from brainscore_vision.model_helpers.activations.pytorch import load_preprocess_images

def get_model():
    model = torchvision.models.mymodel(pretrained=True)
    preprocessing = functools.partial(load_preprocess_images, image_size=224)
    wrapper = PytorchWrapper(identifier='mymodel', model=model, preprocessing=preprocessing)
    wrapper.image_size = 224
    return wrapper

activations_model=get_model()

mybrainmodel = ModelCommitment(
    identifier=activations_model.identifier,
    activations_model=activations_model,
    layers=mylayers)
```

**Comment by YudiXie** [comment]
Yes, the `local_score` could be very simple. 

The concerns I had when first doing something similar to `local_score` is that:
1. It bypassed some new functions such as `ImportPlugin` and `CondaScore`, as a user I don't know what these are doing, so without confirmation that something like `local_score` could work, I am unsure this would give me the correct result, and it feels a little hacky.
2. It bypasses the `save_score` function in brainscore, so the user has to save the score separately, and when rerunning the same benchmark on the same model, some computation has to be rerun. Since brainscore already implemented some function to save/cache the score, it would nice if `local_score` could reuse these. (this is just a small suggestion to further improve the user experience and it is unrelated to the core function, I understand there may be other considerations that make this not necessary now.)

Some examples about how to wrap the models and score them locally using `local_score` in the tutorial would be awesome! They would show the user this is also a supported use case (if the team wants it to be), so that they won't be unsure about whether they did the right thing or not.

**Comment by niranjanrajesh02** [comment]
Hey, just wanted to check if there is any documentation or new functionality to score models locally? 
I am planning to run some pretty large CNNs, so I also wanted to know if there is an optimized way to handle this or do I just choose a few layers to possibly commit?

**Comment by KartikP** [comment]
Hi @niranjanrajesh02 . We are planning on releasing some updated Jupyter notebooks that should provide some clarity on this. These will be available on the website under Tutorials. We're currently working on optimizing some of the backend and revamping the leaderboard at the moment, and the next thing will be making these notebooks available.

As for scoring models locally, the discussion above and the notebook mentioned still works and is a supported way to score models locally. Additionally, you can refer to #533 , #1655 and #1297 for additional information on how to handle layer commitment.

**Comment by niranjanrajesh02** [comment]
Hey @KartikP , great to hear that more documentation and examples are on the way! I look forward to them.

Thank you, I was able to get local model scoring working well! I am attaching a minimal example here that combines the code from this discussion above that may be helpful to others:

```python
from brainscore_vision.model_helpers.brain_transformation import ModelCommitment
from brainscore_vision.model_helpers.activations.pytorch import load_preprocess_images, PytorchWrapper
from brainscore_vision import load_benchmark, BrainModel, Score
import functools

def get_model(model_name):
    model = load_model(model_name) # load a pytorch model
    preprocessing = functools.partial(load_preprocess_images, image_size=224)
    wrapper = PytorchWrapper(identifier=model_name, model=model, preprocessing=preprocessing)
    wrapper.image_size = 224
    return wrapper

def local_score(model: BrainModel, benchmark_identifier: str) -> Score:
    benchmark = load_benchmark(benchmark_identifier)
    score = benchmark(model)
    return score

def main():
    model_name = 'vgg16'
    activations_model=get_model(model_name)

    mybrainmodel = ModelCommitment(
        identifier=activations_model.identifier,
        activations_model=activations_model,
        layers=['head.global_pool.flatten']) # need to seperate modules with a '.'
    
    myscore = local_score(model=mybrainmodel, benchmark_identifier='MajajHong2015public.IT-pls')

    print(f'Model: {model_name}, Score: {myscore.item()}')
```
